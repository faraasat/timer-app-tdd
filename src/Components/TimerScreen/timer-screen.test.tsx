import { shallow } from "enzyme";
import TimerScreenComponent from "./timer-screen.component";

describe("Timer Component Shallow Render", () => {
  const mockStates = {
    playState: false,
    pauseState: false,
    resetState: false,
    setResetState: function abc() {},
  };

  const shallowRender = shallow(
    <TimerScreenComponent
      playState={mockStates.playState}
      pauseState={mockStates.pauseState}
      resetState={mockStates.resetState}
      setResetState={mockStates.setResetState}
    />
  );

  it("should match snapshot", () => {
    expect(shallowRender.debug()).toMatchSnapshot();
  });

  it("should render a div", () => {
    expect(shallowRender.find("div").length).toBeGreaterThanOrEqual(1);
  });

  it("should render 5 spans", () => {
    expect(shallowRender.find("span").length).toEqual(4);
  });

  it("should have a className timer-screen", () => {
    expect(shallowRender.find(".timer-screen").length).toBe(1);
  });

  it("should have a className timer-screen__symbol", () => {
    expect(shallowRender.find(".timer-screen__symbol").length).toBe(3);
  });

  it("should have a className timer-screen__display", () => {
    expect(shallowRender.find(".timer-screen__display").length).toBe(1);
  });
});
