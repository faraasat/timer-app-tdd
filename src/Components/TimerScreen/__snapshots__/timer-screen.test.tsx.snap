// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Component Shallow Render should match snapshot 1`] = `
"<div className=\\"timer-screen\\">
  <span className=\\"timer-screen__display\\">
    00
     
    <span className=\\"timer-screen__symbol\\">
      :
    </span>
     
    00
     
    <span className=\\"timer-screen__symbol\\">
      :
    </span>
     
    00
     
    <span className=\\"timer-screen__symbol\\">
      :
    </span>
     
    00
  </span>
</div>"
`;
